crates/uv/src/commands/pip/operations.rs

```diff
@@ -50,13 +50,33 @@ use crate::printer::Printer;
 
 /// Consolidate the requirements for an installation.
 pub(crate) async fn read_requirements(
-    requirements: &[RequirementsSource],
+    mut requirements: &[RequirementsSource],
     constraints: &[RequirementsSource],
     overrides: &[RequirementsSource],
     extras: &ExtrasSpecification,
-    groups: &DevGroupsSpecification,
+    groups: &[PipGroupName],
     client_builder: &BaseClientBuilder<'_>,
 ) -> Result<RequirementsSpecification, Error> {
+    // pip `--group` flags specify their own sources, and basically disable everything else.
+    // So if we encounter them, we desugar them to `-r` inputs and proceed as normal.
+    let group_requirements;
+    if !groups.is_empty() {
+        debug_assert!(
+            requirements.is_empty(),
+            "-r should be exclusive with --group in `uv pip`"
+        );
+        // Deduplicate in a stable way to get deterministic behaviour
+        let deduped_paths = groups
+            .iter()
+            .map(|group| &group.path)
+            .collect::<BTreeSet<_>>();
+        group_requirements = deduped_paths
+            .into_iter()
+            .map(|path| RequirementsSource::PyprojectToml(path.to_owned()))
+            .collect::<Vec<_>>();
+        requirements = &group_requirements[..];
```

Comment:
I started doing a cleanup pass to compute a `BTreeMap<PathBuf, Vec<GroupName>>` that would be used by subsequent steps but it was just more code and value threading for no real benefit over the current impl that just does a `filter(group.path == path)` later.